torch/jit/__init__.py

```diff
@@ -1774,34 +1728,28 @@ def __init__(self, optimize=True):
             super(ScriptModule, self).__init__()
 
 
-def _get_weak_stubs(cls):
-    """
-    Calls script_method for each method that has been annotated with @weak_script
-    on the type of the object passed in and returns the generated ScriptMethodStubs.
-    """
-    stubs = []
-    for name in dir(cls):
-        func = get_function_from_type(cls, name)
-        if func in _jit_internal.weak_script_methods:
-            entry = _jit_internal.weak_script_methods[func]
-            stub = script_method(entry["original_method"], entry["rcb"])
-            stubs.append(stub)
-    return stubs
-
-
 def _convert_to_script_module(mod, methods=None):
     """
     Makes a ScriptModule from an nn.Module. If `_methods` is provided,
     these methods are treated as @script_methods. If not, it defaults to
     `('forward',)`. Methods accessed in forward are scripted on demand if
     `_enable_recursive_script()` is used.
     """
+    if isinstance(mod, ScriptModule):
+        return mod
+
     if isinstance(mod, (ModuleList, Sequential)):
         # Create constant versions for the iterable modules
         return _create_constant_iterable_module(mod)
 
-    if methods is None:
-        methods = ('forward',)
+    if methods is None and hasattr(mod, 'forward'):
+        if mod.forward.__func__ == torch.nn.Module.forward:
+            # forward was not overrided
+            raise RuntimeError("No forward (TODO: delete this error)")
+        if not _jit_internal.is_ignored_fn(mod.forward):
```

Comment:
I think we should delete the `methods` argument. It is never used in the current codebase.